internal/testplanet/uplink.go

```diff
@@ -342,75 +332,94 @@ func atLeastOne(value int) int {
 	return value
 }
 
-// DialMetainfo returns a metainfo and streams store for the given configuration and identity.
-func DialMetainfo(ctx context.Context, log *zap.Logger, config uplink.Config, identity *identity.FullIdentity) (db storj.Metainfo, ss streams.Store, cleanup func() error, err error) {
-	tlsOpts, err := tlsopts.NewOptions(identity, config.TLS)
-	if err != nil {
-		return nil, nil, cleanup, err
+// NewLibuplink creates a libuplink.Uplink object with the testplanet Uplink config
+func (planetUplink *Uplink) NewLibuplink(ctx context.Context) (*libuplink.Uplink, error) {
+	config := getDefaultConfig()
+	libuplinkCfg := &libuplink.Config{}
+	libuplinkCfg.Volatile.MaxInlineSize = config.Client.MaxInlineSize
+	libuplinkCfg.Volatile.MaxMemory = config.RS.MaxBufferMem
+	libuplinkCfg.Volatile.PeerIDVersion = config.TLS.PeerIDVersions
+	libuplinkCfg.Volatile.TLS = struct {
+		SkipPeerCAWhitelist bool
```

Comment:
This is a nitpick, but this pattern seems less clear to me than simply continuing what you're doing above, such as:

```
libuplinkCfg.Volatile.TLS.SkipPeerCAWhitelist = !config.TLS.UsePeerCAWhitelist
libuplinkCfg.Volatile.TLS.PeerCAWhitelistPath = config.TLS.PeerCAWhitelistPath
```

Reply:
sounds good. yeah i really dislike instantiating this object like this. its still bad, but your way is prob a tad better. will update